core/vm/actor/builtin/v0/payment_channel/payment_channel_actor.cpp

```diff
@@ -78,10 +78,12 @@ namespace fc::vm::actor::builtin::v0::payment_channel {
     if (voucher.amount.sign() < 0) {
       return VMExitCode::kErrIllegalArgument;
     }
-    if (!voucher.secret_preimage.empty()
-        && gsl::make_span(runtime.hashBlake2b(params.secret))
-               != gsl::make_span(voucher.secret_preimage)) {
-      return VMExitCode::kErrIllegalArgument;
+    if (!voucher.secret_preimage.empty()) {
+      OUTCOME_TRY(hash, runtime.hashBlake2b(params.secret));
+      if (gsl::make_span(std::move(hash))
```

Comment:
```suggestion
      if (gsl::make_span(hash)
```

Reply:
It could not be compiled, because 2 different make_span() functions are called - const and non-const. And there is no operator== for const and non-const values.
If you want to avoid moving, I just create a copy of voucher.secret_preimage.